library.c

```diff
@@ -333,16 +333,16 @@ redis_check_eof(RedisSock *redis_sock, int no_throw)
         errmsg = "Connection lost and socket is in MULTI/watching mode";
     } else {
         errmsg = "Connection lost";
-        /* TODO: configurable max retry count */
-        for (count = 0; count < 10; ++count) {
+        backoff_reset(&redis_sock->backoff);
+        for (retry_index = 0; retry_index < redis_sock->max_retries; ++retry_index) {
             /* close existing stream before reconnecting */
             if (redis_sock->stream) {
                 redis_sock_disconnect(redis_sock, 1);
             }
             // Wait for a while before trying to reconnect
             if (redis_sock->retry_interval) {
                 // Random factor to avoid having several (or many) concurrent connections trying to reconnect at the same time
-                long retry_interval = (count ? redis_sock->retry_interval : (php_rand() % redis_sock->retry_interval));
+                zend_long retry_interval = backoff_compute(&redis_sock->backoff, retry_index);
                 usleep(retry_interval);
             }
```

Comment:
Maybe we can remove the `if (retry_interval) { ... }` and just compute the backoff with `backoff_compute`.  Simpler to follow I think.

```c
            /* Sleep based on our backoff algorithm */
            zend_long delay = backoff_compute(&redis_sock->backoff, retry_index);
            if (delay != 0)
                usleep(delay);
```

From my initial testing this works as expected (e.g. returns 0 if retry_interval is 0)

Reply:
Agreed. Fixed.